posydon/utils/common_functions.py

```diff
@@ -195,52 +195,54 @@ def rzams(m, z=0.02, Zsun=0.02):
     return r
 
 
-'''
-
-
-Receives:
-q ->
-a_orb ->
-
-Returns:
-RL -> Roche lobe radius in similar units as a_orb
-'''
-
-
 def roche_lobe_radius(m1, m2, a_orb=1):
     """Approximate the Roche lobe radius from [1]_.
 
     Parameters
     ----------
-    m1 : float
+    m1 : float, array of floats
         the mass of the star for which we calculate the Roche lobe
-    m2: float
+    m2 : float, array of floats
         the mass of the companion star
-    a_orb : float
+    a_orb : float, array of floats
         Orbital separation. The return value will have the same unit.
 
     Returns
     -------
-    float
+    float, array of floats
         Roche lobe radius in similar units as a_orb
     References
     ----------
     .. [1] Eggleton, P. P. 1983, ApJ, 268, 368
 
     """
-
-    if isinstance(a_orb, np.ndarray):
+    
+    ## catching if a_orb is an empty array or is an array with invalid separation values
+    if isinstance(a_orb, np.ndarray) and (not np.any(a_orb) or np.any(a_orb <=0)):
```

Comment:
Can you remind me, what `not np.any(a_orb)` is good for? Hence, what is covered by that, that's not covered by `np.any(a_orb <=0)`? I guess, you look for an empty array, but there you should use `len(a_orb)==0`, which should be more efficient because it doesn't need to touch all the values stored in the ndarray, additionally `np.any` would fail if 0 would be an allowed value.
Instead, we should have a check on the dtype of the ndarray.

Reply:
Yes, np.any() will check for an empty array. I would not use len() in this case because it will return the number of dimensions for a numpy array if the array has more than one dimension, which is not what we want. 
np.any() should not fail if one of the entries is zero, I do not understand what you mean there. Do you mean we want to allow for the orbital period to equal zero?

Reply:
I think a_orb.size will do this check better, I am changing to that

Reply:
Yes, using the `size` is what I intended when suggesting `len`. `np.any` has always the issue, that is returns `False` on both an empty ndarray and an ndarray containing only values which are convertible to False individually.